### PR TITLE
USWDS-Site - Documentation: Add `$theme-utility-breakpoints-custom` to settings table.

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

Added `$theme-utility-breakpoints-custom` setting to the utility settings table.

| Name | var | default | type | description |
|--------|--------|--------|--------|--------|
| Custom breakpoints | `$theme-utility-breakpoints-custom` |  | map | Set additional custom mobile-first breakpoints to be output as utilities alongside the enabled system breakpoints. Values must be passed in a scss map as px. | 

## Related PR

Documentation for uswds/uswds#6048

## Preview link

[Settings page →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-custom-breakpoint-documentation/documentation/settings/#utilities-settings-2)

## Major changes

- Added the new theme setting to the utility settings table.

## Testing and review

1. Visit [settings page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-custom-breakpoint-documentation/documentation/settings/#utilities-settings-2)
2. Read new setting description
3. Confirm description has appropriate name, var, default, type, and description values
4. Confirm placement for the new setting is appropriate

### CODE REVIEW

File: _data/changelogs/docs-settings.yml [hunk 1]
```diff
@@ -2,6 +2,14 @@ title: Settings
 type: utility
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Added new `$theme-utility-breakpoints-custom` setting.
+    summaryAdditional:  Users can now set custom breakpoints which will generate utilities alongside the default system breakpoints.
```
Comment (amyleadem) [inline _data/changelogs/docs-settings.yml:7]: **suggestion**:  Some minor tweaks to improve readability
```suggestion
    summaryAdditional:  Users can use this new setting to generate responsive variants of USWDS utilities at custom breakpoints. 
```

File: _data/settings/utilities.yml [hunk 2]
```diff
@@ -60,6 +60,13 @@ contents:
         value: "false"
         type: boolean
 
+  - name: Custom breakpoints
+    description: |
+      Set additional custom mobile-first breakpoints to be output as utilities alongside the enabled system breakpoints. Values must be passed in a scss map as px.
```
Comment (amyleadem) [inline _data/settings/utilities.yml:65]: **suggestion**: Small tweaks to align with description for `$theme-utility-breakpoints` and to improve clarity. 
```suggestion
      Set additional custom mobile-first breakpoints to be output for the utilities. Values must be passed in with `px` values inside a Sass map.
```

**Comment by amyleadem** [comment]
**suggestion**: 
I am thinking that we should add some additional notes about this setting to [responsive-variants.html](https://github.com/uswds/uswds-site/blob/main/_includes/utilities/responsive-variants.html). 

1. Suggesting we update the section to include a sentence like this:
"Additionally, you can add custom breakpoints to your project by configuring the `$theme-utility-breakpoints-custom` setting. Learn more about the breakpoint settings on the [Settings page](https://designsystem.digital.gov/settings/#utilities-settings)."
    
    Before:
    image

    After:
    image

2. Additionally, it might be a good opportunity to add a yellow site note here that warns users that each breakpoint that is turned on makes a sizeable impact to the compiled CSS. (right now, adding a new breakpoint adds about 50kb to the total file size). Here is a possible draft note (this will probably need to be cleaned up):

    "**Note:** Turn on only the breakpoints you need in your project. Each enabled breakpoint will increase the size of your compiled CSS by a significant amount. To improve performance on your site, turn off any unused breakpoints."

File: _data/changelogs/docs-settings.yml (hunk 1 above)
Comment (mejiaj) [reply _data/changelogs/docs-settings.yml:7]: @amyleadem I'd argue that the original is a little clearer. First, it mentions that this is a new setting and second it tells you that new utils can be generated from it.

**Review by mejiaj — APPROVED** [review]
Changelog looks good and added suggestion for clarity in [settings table](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-custom-breakpoint-documentation/documentation/settings/#utilities-settings-2:~:text=Set%20additional%20custom%20mobile%2Dfirst%20breakpoints%20to%20be%20output%20as%20utilities%20alongside%20the%20enabled%20system%20breakpoints.%20Values%20must%20be%20passed%20in%20a%20scss%20map%20as%20px.).

**Comment by amyleadem** [comment]
Update: I'm going to move [my suggestions to enhance responsive-variants.html](https://github.com/uswds/uswds-site/pull/2837#issuecomment-2384318751) to a separate PR so that it doesn't hold up the release. We can evaluate it independently as a follow-on if needed

**Pull request merged**